src/components/home/components/actions.tsx

```diff
@@ -0,0 +1,74 @@
+import { Box, Button, ButtonProps, Stack, StackProps } from '@stacks/ui';
+import { ScreenPaths } from '@store/common/types';
+import React, { memo, useCallback, useRef } from 'react';
+import { useDoChangeScreen } from '@common/hooks/use-do-change-screen';
+import { IconArrowUp, IconQrcode } from '@tabler/icons';
+import { useTransferableAssets } from '@common/hooks/use-assets';
+
+interface TxButtonProps extends ButtonProps {
+  kind: 'send' | 'receive';
+  path: ScreenPaths.POPUP_SEND | ScreenPaths.POPUP_RECEIVE;
+}
+
+const TxButton: React.FC<TxButtonProps> = memo(({ kind, path, ...rest }) => {
+  const ref = useRef<HTMLButtonElement | null>(null);
+  const doChangeScreen = useDoChangeScreen();
+
+  const isSend = kind === 'send';
+
+  const handleClick = useCallback(() => {
+    doChangeScreen(path);
+  }, [path, doChangeScreen]);
+
+  const label = isSend ? 'Send' : 'Receive';
+  return (
+    <>
+      <Button
+        size="sm"
+        pl="base-tight"
+        pr={'base'}
+        py="tight"
+        fontSize={2}
+        mode="primary"
+        position="relative"
+        ref={ref}
+        onClick={handleClick}
+        borderRadius="12px"
+        {...rest}
+      >
+        <Box
+          as={isSend ? IconArrowUp : IconQrcode}
+          transform={isSend ? 'unset' : 'scaleY(-1)'}
+          size="16px"
+        />
+        <Box as="span" ml="extra-tight" fontSize="14px">
+          {label}
+        </Box>
+      </Button>
+    </>
+  );
+});
+
+const SendButtonSuspense = () => {
+  const assets = useTransferableAssets();
+  const isDisabled = !assets || assets?.length === 0;
+  return <TxButton isDisabled={isDisabled} path={ScreenPaths.POPUP_SEND} kind="send" />;
+};
+const SendButtonFallback = memo(() => (
+  <TxButton isDisabled path={ScreenPaths.POPUP_SEND} kind="send" />
+));
```

Comment:
So I think the reason I'd prefer another naming convention is that, the way I see it, hooks introduce a way with which components can use functionality, without being concerned with its implementation.

The fact that `useTransferableAssets()` suspends is an implementation detail that the component is concerned with, because the name includes it.

I reckon it's more helpful for the naming to describe where the suspense boundary and its fallback are, than being aware of one of the hook's implementation details.

Reply:
Can you give an example of what makes most sense to you?

Reply:
Here's an attempt: I'm not 100% on these, there may be better alternatives, but they address the above in not making reference to the containing hook's implementation:

`<SendButton />` --> base component describing what the component _is_, dumb
`<SendAction />` --> component describing what the component _does_, higher level (doesn't reference specific UI element used), smart, contains logic
`<SendActionWithFallback />` --> component responsible for describing suspense and fallback case 🤷🏼‍♂️ 

I wonder if there's an argument for whether we _should_ create separate components with the suspense in. In this instance for example, couldn't it be considered the role of `HomeActions` to provide a fallback for the given component? 🤔

Reply:
yeah i agree with a lot of what you're saying and am very open to finding something else. i think it's something we'll have to work through to really nail down imo, esp as it's pretty new. I think there are arguments for having the async component be used elsewhere, and as such the component implementing it might want a different or no fallback, but elsewhere we will want to be able to put a component somewhere and it will handle its fallback _and_ error state. https://reactjs.org/docs/concurrent-mode-suspense.html#handling-errors was just reading this, as an example.

I think I'll make a discussion so we have a place that isn't a PR comment to really dig into this, and get others opinions, would that work?

Reply:
Yep makes sense 👍🏼